ui/dashboards/src/context/DashboardAppSlice.tsx

```diff
@@ -0,0 +1,45 @@
+import { immer } from 'zustand/middleware/immer';
+import { useDashboardStore } from './DashboardProvider';
+
+export interface DashboardAppSlice {
+  addPanelComponent: {
+    isOpen: boolean;
+    groupIndex: number;
+    setIsOpen: (isOpen: boolean, groupIndex?: number, panelRef?: string) => void;
+    panelRef?: string;
+  };
+  addGroupComponent: {
+    isOpen: boolean;
+    setIsOpen: (isOpen: boolean, index?: number) => void;
+    index?: number;
+  };
+}
+
+export const createDashboardAppSlice = immer<DashboardAppSlice>((set) => ({
+  addPanelComponent: {
+    isOpen: false,
+    groupIndex: 0,
+    setIsOpen: (isOpen: boolean, groupIndex = 0, panelRef?: string) =>
```

Comment:
Since the add panel component is used for adding new panel and editing existing one, we need know when to open Add Panel vs Edit Panel.

if `panelRef` is undefined, it means we're adding a new panel